azext_edge/edge/providers/support/mq.py

```diff
@@ -47,7 +48,28 @@ def fetch_diagnostic_metrics(namespace: str):
             "zinfo": f"{namespace}/{MQ_ACTIVE_API.moniker}/diagnostic_metrics.txt",
         }
     except Exception:
-        logger.debug(f"Unable to call stats pod metrics against namespace {namespace}.")
+        logger.debug(f"Unable to process diagnostics pod metrics against namespace {namespace}.")
+
+
+def fetch_diagnostic_traces():
+    namespaces = get_mq_namespaces()
+    result = []
+    for namespace in namespaces:
+        try:
+            traces = get_traces(namespace=namespace, trace_ids=["!support_bundle!"])
+            if traces:
```

Comment:
nit - now that `get_traces` can return either a list of dicts or a list of tuples, do we need to typecheck its return value? Might even just be here, as the other reference to it is a straight `return get_traces(...`

Reply:
Added test coverage for the return value in the stats tests area.

If the wrong type gets returned, and the later accessing code here causes an exception it should get caught, and also, we have a backup exception handler at the support bundle runner level.

Reply:
Also thinking about the type check it would per list item since the outer most type is either list or None

Reply:
Let's discuss in subsequent PR.